### PR TITLE
Country Weights, Extreme Reshaping redone.

### PR DESCRIPTION
**Pull request by Zemurin**

### CODE REVIEW

**Review by IhateTrains — APPROVED** [review]

File: EU4toV2/Source/V2World/V2World.cpp [hunk 1]
```diff
@@ -1044,7 +1044,7 @@ std::optional<std::string> V2::World::determineProvinceOwnership(const std::vect
 	}
 	if (winner.empty())
 		return std::nullopt;
-	return winner;
+	return std::move(winner);
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/V2World.cpp:1047]: In general, you shouldn't do a std::move on return values, as the compiler optimizes that for you (indeed, its required to do so).

File: EU4toV2/Source/V2World/Province/Province.cpp [hunk 2]
```diff
@@ -354,7 +352,7 @@ std::optional<std::shared_ptr<V2::Factory>> V2::Province::addFactory(std::shared
 	if (itr == factories.end())
 	{
 		factories.insert(std::make_pair(factory->getTypeName(), factory));
-		return factory;
+		return std::move(factory);
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/Province/Province.cpp:355]: As mentioned elsewhere, not need for the std::move

File: EU4toV2/Source/V2World/V2World.cpp [hunk 3]
```diff
@@ -1073,7 +1073,7 @@ std::optional<std::string> V2::World::determineProvinceControllership(const std:
...
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/V2World.cpp:1076]: As mentioned elsewhere, not need for the std::move

**Review by Idhrendur — APPROVED** [review]
Deleted code a simplified interfaces make me happy.

**Pull request merged**